exchange/bidder.go

```diff
@@ -331,10 +331,16 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 							adjustmentFactor = givenAdjustment
 						}
 
+						adjArray := []openrtb_ext.Adjustments{}
+						if bidAdjustments != nil {
+							adjArray = bidAdjustments.GetAdjustmentArray(bidResponse.Bids[i].BidType, bidderRequest.BidderName, bidResponse.Bids[i].Bid.DealID)
+						}
+
 						originalBidCpm := 0.0
 						if bidResponse.Bids[i].Bid != nil {
 							originalBidCpm = bidResponse.Bids[i].Bid.Price
 							bidResponse.Bids[i].Bid.Price = bidResponse.Bids[i].Bid.Price * adjustmentFactor * conversionRate
+							bidResponse.Bids[i].Bid.Price = applyAdjustmentArray(adjArray, bidResponse.Bids[i].Bid.Price, bidResponse.Currency, reqInfo)
```

Comment:
I like that idea! I think I'd want to bring it up during team time first before going through with this :)

Reply:
I realized after thinking about this, and I think we need to keep these two functions separate. The `GetAdjustmentArray` function is needed on it's own to satisfy the requirement of: "Any module that needs to reverse the adjustment (e.g. floors), will need a way to get the relevant adjustment array and they can walk the adjustments backwards. This implies that there should be a function similar to the Floors getFloor() function that allows callers to supply dimension values and get the adjustments array."

Reply:
You can still have `GetAdjustmentArray` as a separate function and call it first in the new function that will consolidate 
`GetAdjustmentArray` and `applyAdjustmentArray`. It seems like a logical way to do this, it will minimize functions calls from `requestBid`, it's already complicated. And also `adjArray` is only used in `applyAdjustmentArray`.